### PR TITLE
Add option to build offline HTML

### PR DESCRIPTION
**Pull request by lunaryorn**
With this PR `make -C doc OFFLINE=yes html` creates HTML that does not load any external resources.  See #999 for background.

/cc @spwhitton 

@cpitclaudel Please take a look.

### CODE REVIEW

**Comment by lunaryorn** [comment]
@spwhitton Please take a look at this change.  I'd be awesome if you could test it and see whether the documentation generated with this change suits your needs.

**Comment by lunaryorn** [comment]
For the bot: LGTM.

File: doc/_templates/about.html [hunk 1]
```diff
@@ -19,7 +19,7 @@ <h1 class="logo"><a href="{{ pathto(master_doc) }}">{{ project }}</a></h1>
 <p><a href="{{href}}"><img src="https://img.shields.io/{{badge}}"/></a></p>
 {%- endmacro %}
 
-{% if theme_github_user and theme_github_repo %}
+{% if theme_github_user and theme_github_repo and not flycheck_offline_html %}
 {% if theme_github_button|lower == 'true' %}
 <p>
     <iframe src="https://ghbtns.com/github-btn.html?user={{ theme_github_user }}&repo={{ theme_github_repo }}&type={{ theme_github_type }}&count={{ theme_github_count }}&v=2"
```
Comment (cpitclaudel) [inline doc/_templates/about.html]: Shouldn't these be `&amp;` instead of `&`?

Comment (lunaryorn) [inline doc/_templates/about.html]: Uh, uhm, maybe :wink:

Comment (lunaryorn) [inline doc/_templates/about.html]: Fixed 😎

**Comment by spwhitton** [comment]
LGTM!  Thanks!

**Comment by lunaryorn** [comment]
@cpitclaudel Do you approve?

**Comment by cpitclaudel** [comment]
Yes, LGTM :)
(I thought I did back when I reviewed the diff, sorry!)

**Comment by lunaryorn** [comment]
@spwhitton Please feel free to squash-merge 😊

**Comment by lunaryorn** [comment]
@spwhitton Thanks, also for deleting the branch 😊

**Pull request merged**